x/crosschain/keeper/msg_server_vote_inbound_tx_test.go

```diff
@@ -160,6 +162,64 @@ func TestKeeper_VoteOnObservedInboundTx(t *testing.T) {
 		_, found = zk.ObserverKeeper.GetBallot(ctx, msg2.Digest())
 		require.False(t, found)
 	})
+
+	t.Run("should error if vote on inbound ballot fails", func(t *testing.T) {
+		k, ctx, _, _ := keepertest.CrosschainKeeperWithMocks(t, keepertest.CrosschainMockOptions{
+			UseObserverMock: true,
+		})
+		observerMock := keepertest.GetCrosschainObserverMock(t, k)
+		observerMock.On("VoteOnInboundBallot", mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything).
+			Return(true, false, errors.New("err"))
+		msgServer := keeper.NewMsgServerImpl(*k)
+		to, from := int64(1337), int64(101)
+
+		msg := sample.InboundVote(0, from, to)
+		res, err := msgServer.VoteOnObservedInboundTx(
+			ctx,
+			&msg,
+		)
+		require.Error(t, err)
+		require.Nil(t, res)
+	})
+
+	t.Run("should return if not finalized", func(t *testing.T) {
+		k, ctx, _, _ := keepertest.CrosschainKeeperWithMocks(t, keepertest.CrosschainMockOptions{
+			UseObserverMock: true,
```

Comment:
do you think it makes sense to not use a mock here and check if the vote gets added.